File: recursion/python/iteration.py
def factorial(n):
    result = 1
    for i in range(n):
        result *= i+1
    return result


def main():
    test_cases = int(input())
    for _ in range(test_cases):
        n = int(input())
        print(factorial(n))


# Expected:
# 1
# 720 -> 6 * 5 * 4 * 3 * 2 * 1
# 5040

if __name__ == "__main__":
    main()

File: recursion_basics/cpp/iteration.cpp
#include <iostream>

using namespace std;

int factorial(int n) {
    int result = 1;
    for (int i = 0; i < n; ++i) {
        result *= i+1;
    }
    return result;
}

int main() {
    int test_cases; cin >> test_cases;
    for (int tc = 0; tc < test_cases; ++tc) {
        int n; cin >> n;
        cout << factorial(n) << endl;
    }
}